itests/migration_test.go

```diff
@@ -894,6 +900,243 @@ func TestMigrationNV23(t *testing.T) {
 	require.NoError(t, err)
-	state := f090StatePost.State.(*account.State)
+	state := f090StatePost.State.(*account14.State)
 	require.Equal(t, state.Address, f090Addr)
 }
+
+func TestMigrationNV24(t *testing.T) {
+	req := require.New(t)
+
+	kit.QuietMiningLogs()
+
+	const (
+		nv24epoch               abi.ChainEpoch = 100
+		powerRampDurationEpochs uint64         = 200
+		blockTime                              = 10 * time.Millisecond
+	)
+	buildconstants.UpgradeTuktukPowerRampDurationEpochs = powerRampDurationEpochs
+
+	// InitialPledgeMaxPerByte is a little too low for an itest environment so gets in the way of testing the underlying calculation
+	miner14.InitialPledgeMaxPerByte = big.Mul(miner14.InitialPledgeMaxPerByte, big.NewInt(10)) // pre migration
+	miner15.InitialPledgeMaxPerByte = big.Mul(miner15.InitialPledgeMaxPerByte, big.NewInt(10)) // post migration
+
+	// Observe the rate of change of the pledge calculation during and after the power ramp; change
+	// is measured as a difference between the pre-FIP-0081 pledge calculation and the calculation
+	// after FIP-0081 has been applied.
+	var (
+		rateOfChangeDuringRamp []float64
+		rateOfChangeAfterRamp  []float64
+	)
+
+	testClient, _, ens := kit.EnsembleMinimal(
+		t,
+		kit.MockProofs(),
+		kit.UpgradeSchedule(stmgr.Upgrade{
+			Network: network.Version23,
+			Height:  -1,
+		}, stmgr.Upgrade{
+			Network:   network.Version24,
+			Height:    nv24epoch,
+			Migration: filcns.UpgradeActorsV15,
+		},
+		))
+
+	ens.InterconnectAll().BeginMining(blockTime)
+
+	clientApi := testClient.FullNode.(*impl.FullNodeAPI)
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	// Watch the chain and capture the pledge every ~5 epochs to observe the rate of change before,
+	// during, and after the FIP-0081 pledge ramp.
+	doneCh := make(chan struct{})
+	go func() {
+		var lastRelativeChange float64
+		var lastEpoch abi.ChainEpoch
+
+		start, err := clientApi.ChainHead(ctx)
+		assert.NoError(t, err)
+
+		for i := 0; ; i++ {
+			head := testClient.WaitTillChain(ctx, kit.HeightAtLeast(start.Height()+abi.ChainEpoch(i*5)))
+			if i > 0 && head.Height() == lastEpoch {
+				t.Logf("skipping duplicate pledge calculation @%d, slow test runner?", head.Height())
+				continue
+			}
+			lastEpoch = head.Height()
+
+			pledge, err := clientApi.StateMinerInitialPledgeForSector(ctx, abi.ChainEpoch(builtin.EpochsInYear), abi.SectorSize(2<<10), 0, head.Key())
+			if err != nil {
+				t.Errorf("failed to calculate pledge %d: %s", head.Height(), err)
+				break
+			}
+			preFip0081Pledge := preFip0081StateMinerInitialPledgeForSector(ctx, t, clientApi, abi.ChainEpoch(builtin.EpochsInYear), abi.SectorSize(2<<10), 0, head.Key())
+
+			relativeChange := float64(pledge.Uint64()-preFip0081Pledge.Uint64()) / float64(preFip0081Pledge.Uint64())
+			if i > 0 {
+				if head.Height() > nv24epoch {
+					// We want to see an increasing distance between the pre-fip-0081 pledge and original pledge
```

Comment:
Its worth documenting somewhere that this test relies on network power being below baseline.  The impact is in the opposite direction if we are well above baseline

Reply:
My mistake -- not opposite direction but no change if above baseline

Reply:
👌 good point, documented right here